client/src/modules/TeamDistribution/components/TeamDistributionModal/TeamDistributionModal.tsx

```diff
@@ -75,97 +79,85 @@ const submitTeamDistribution = async (courseId: number, values: Partial<FormStat
 };
 
 export default function TeamDistributionModal({ data, onCancel, courseId, onSubmit }: Props) {
-  const [form] = Form.useForm<Partial<FormState>>();
-  const [changes, setChanges] = useState<Record<string, boolean>>({
-    strictTeamSizeMode: data.strictTeamSizeMode ?? true,
-  });
-
-  const handleModalSubmit = async (values: Partial<FormState>) => {
-    await submitTeamDistribution(courseId, values, data.id);
+  const [form] = Form.useForm<FormState>();
+  const handleModalSubmit = async (values: FormState) => {
+    await submitTeamDistribution(courseId, values, data?.id);
     await onSubmit();
   };
 
   return (
-    <ModalForm
-      form={form}
-      data={data}
-      onChange={values => setChanges({ strictTeamSizeMode: values.strictTeamSizeMode })}
-      title="Team Distribution"
-      submit={handleModalSubmit}
-      cancel={onCancel}
-      getInitialValues={getInitialValues}
+    <Modal
+      open={true}
+      title={'Team distribution'}
+      width={756}
+      onOk={async () => {
+        const values = await form.validateFields().catch(() => null);
+        if (values == null) {
+          return;
+        }
+        handleModalSubmit(values);
+      }}
+      onCancel={() => {
+        onCancel();
+        form.resetFields();
+      }}
     >
-      <Form.Item name="name" label="Name" rules={[{ required: true, message: 'Please enter event name' }]}>
-        <Input />
-      </Form.Item>
-      <Row gutter={24}>
-        <Col span={18}>
-          <Form.Item
-            name="range"
-            label="Start Date - End Date"
-            rules={[{ required: true, type: 'array', message: 'Please enter start and end date' }]}
-          >
-            <DatePicker.RangePicker
-              format="YYYY-MM-DD HH:mm"
-              showTime={{ format: 'HH:mm', defaultValue: [moment().hour(0).minute(0), moment().hour(23).minute(59)] }}
-            />
-          </Form.Item>
-        </Col>
-        <Col span={6}>
-          <Form.Item name="timeZone" label="TimeZone">
-            <Select placeholder="Please select a timezone">
-              {TIMEZONES.map(tz => (
-                <Option key={tz} value={tz}>
-                  {/* there is no 'Europe / Kyiv' time zone at the moment */}
-                  {tz === 'Europe/Kiev' ? 'Europe/Kyiv' : tz}
-                </Option>
-              ))}
-            </Select>
-          </Form.Item>
-        </Col>
-      </Row>
-      <Form.Item name="strictTeamSizeMode" label="Fixed team size" valuePropName="checked">
-        <Switch />
-      </Form.Item>
-      {changes.strictTeamSizeMode === true ? (
+      <Form {...layout} form={form} initialValues={data ? getInitialValues(data) : undefined}>
+        <Text strong>
+          You are {!data ? 'creating' : 'editing'} a group distribution event. Fill out the form to add it to the
```

Comment:
Maybe we could avoid one operation here
```suggestion
          You are {data ? 'editing' : 'creating' } a group distribution event. Fill out the form to add it to the
```